### PR TITLE
Adding a tor browser test file (just standalone for now)

### PR DESCRIPTION
**Pull request by Hainish**
Ready for review @jsha @cooperq

### CODE REVIEW

File: test/tor-browser.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+source utils/mktemp.sh
+
+die() {
+  echo "$@"
+  exit 1
+}
+
+# Unzip the Tor Browser archive to a temporary directory
+if [ ! -f "$1" ] && [ "`echo $1 | tail -c 8 | head -c 7`" != ".tar.xz" ]; then
+  die "File provided is not a valid Tor Browser archive."
```
Comment (cooperq) [inline test/tor-browser.sh]: Maybe add a usage message here?

File: test/tor-browser.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+then
+    # $GIT_DIR is set, so we're running as a hook.
+    cd $GIT_DIR
+else
+    # Git command exists? Cool, let's CD to the right place.
+    git rev-parse && cd "$(git rev-parse --show-toplevel)"
+fi
+
+
+source utils/mktemp.sh
```
Comment (cooperq) [inline test/tor-browser.sh]: What does this do?

File: test/tor-browser.sh [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+# Build the XPI to run all the validations in makexpi.sh, and to ensure that
+# we test what is actually getting built.
+./makexpi.sh
+XPI_NAME="`ls -tr pkg/*-eff.xpi | tail -1`"
+
+# Install into our fresh Tor Browser
+unzip -qd $HTTPSE_INSTALL_DIRECTORY $XPI_NAME
+
+if [ ! -d "$HTTPSE_INSTALL_DIRECTORY" ]; then
+  die "Tor Browser does not have HTTPS Everywhere installed"
```
Comment (cooperq) [inline test/tor-browser.sh]: Why would this ever happen? If makexpi fails?

**Comment by cooperq** [comment]
lgtm modulo my comments

File: test/tor-browser.sh (hunk 3 above)
Comment (Hainish) [inline test/tor-browser.sh]: Yes, or if the archive is corrupted (the `unzip` step fails)

**Comment by Hainish** [comment]
ready for review again @cooperq

File: test/tor-browser.sh (hunk 2 above)
Comment (Hainish) [inline test/tor-browser.sh]: It just makes `mktemp` available across `GNU/Linux & BSD` systems, apparently.  This is documented in the `utils/mktemp.sh` file.

**Comment by cooperq** [comment]
lgtm

**Pull request merged**